torch/csrc/jit/codegen/cuda/manager.cpp

```diff
@@ -49,29 +61,52 @@ class CudaFusionManager {
 
       graph_cache_[repr] = kernel_id;
 
-      Fusion fusion;
-      // lower torch::jit::Graph to torch::jit::fuser::cuda::fusion
-      parseJitIR(graph, fusion);
-
-      // default constructor via accessing empty key;
-      compileKernel(fusion, kernel_cache_[kernel_id]);
+      // create entry for cached kernel;
+      kernel_cache_.insert({kernel_id, CudaKernelCache()});
 
-      return kernel_id;
-    } else {
-      return graph_cache_[repr];
+      // TODO: we should compile here using profiled information:
+      //       size (range) / stride (contiguity)
     }
+
+    return graph_cache_[repr];
   };
 
   void runFusionNode(
       int32_t kernel_id,
+      std::shared_ptr<Graph>& graph,
       const at::ArrayRef<IValue> inputs,
       std::vector<at::Tensor> outputs) {
-    TORCH_CHECK(
-        kernel_cache_.count(kernel_id) != 0, "kernel id not recognized");
+    std::lock_guard<std::mutex> guard(mutex_);
+    TORCH_CHECK(kernel_cache_.count(kernel_id) != 0, "kernel id not recognized");
+
+    // TODO: temporary hack
+    auto cuda_kernel = kernel_cache_[kernel_id].getKernelPtr(outputs[0].sizes());
+    if (cuda_kernel) {
+      // TODO: update launch config for specific sizes;
+      //       maybe we should store it in CudaKernel and compute it later
+      runKernel(*cuda_kernel, inputs, outputs);
+    } else {
 
-    CudaKernel& cuda_kernel_entry = kernel_cache_[kernel_id];
+      // major HACK!
+      auto kernel_arg_req = expandSizeSupport(outputs[0].sizes());
```

Comment:
What makes it a major hack?

Reply:
because I actually don't have the part implemented yet. :) I should change the comment to a `TODO` instead.